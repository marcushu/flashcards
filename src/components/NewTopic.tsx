import { useState } from 'react';

interface NewTopicProps {
  addNewTopic: (newtopic: string) => void
}

const NewTopic = ({ addNewTopic }: NewTopicProps) => {
  const [newTopicText, setnewTopicText] = useState("");

  const addTopic = () => {
    if (newTopicText === "") {
      alert("Please add a name for your new topic.");
    } else {
      addNewTopic(newTopicText);
      setnewTopicText("");
    }
  }

  return (
    <div>
      <button className="newTopic" onClick={addTopic}>
        +
      </button>
      <input id="newtopicText"
        type="text"
        value={newTopicText}
        placeholder="new topic..."
        onChange={e => setnewTopicText(e.target.value)} />
    </div>
  )
}

export default NewTopic;
