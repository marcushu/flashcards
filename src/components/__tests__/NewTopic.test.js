import React from 'react';
import { fireEvent, render } from '@testing-library/react'
import NewTopic from '../NewTopic.tsx';

describe("NewTopic.test", () => {
    const mockFunc = jest.fn();

    it("renders", () => {
        const { asFragment } = render(
            <NewTopic addNewTopic={mockFunc} />
        );

        expect(asFragment()).toMatchSnapshot();
    });

    it("correctly handles click events", () => {
        const { getByRole } = render(
            <NewTopic addNewTopic={mockFunc} />
        );

        fireEvent.change(getByRole('textbox'), { target: {value: "anytext"}});    
        fireEvent.click(getByRole('button', {name: /\+/i})); 

        expect(mockFunc).toHaveBeenCalled();
        expect(mockFunc).toHaveBeenCalledWith("anytext");
    });
})