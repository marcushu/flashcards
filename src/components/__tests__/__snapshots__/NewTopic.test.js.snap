// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewTopic.test renders 1`] = `
<DocumentFragment>
  <div>
    <button
      id="newTopic"
    >
      +
    </button>
    <input
      id="newtopicText"
      placeholder="new topic..."
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;
